### DISCUSSION TITLE
Trouble with ServerSideEncryption on Minio

### DISCUSSION THREAD

**Discussion by AlexxBlack** [outline/outline#1601] [General]
Good day.
We're using a self-hosted version of Outline, but we were unable to make Outline to work with our s3 bucket.
So we've installed a self-hosted Minio.
Initially everything was well (we upload pics and saw em in the articles). But some time ago we decided to make a different domain name for our company. And replace all user's email by new.
Now they can't even log in to the Outline. Everybody receive such message as: "Authentication failed - we were unable to sign you in at this time. Please try again."
When we are trying to figure out what's happened - we saw an error in server console: "InvalidArgument: Server side encryption specified but KMS is not configured"
Google auth working properly.
So, I suggest that after a success authorization through a Google, Outline try to get user info from server and user avatar from Minio. And this can't be achieved.
So, the question is: How can we turn off a ServerSideEncryption (and remove the x-amz-server-side-encryption header) when outline requests the file storage server (Minio)?
Thank You)

**Comment by tommoor** [comment]
Hi, there is currently no way to turn off server side encryption, it's hardcoded to enabled. This hasn't changed recently so I feel like the question you should be asking is why is the KMS suddenly not configured? This seems separate to moving to a new domain – was configuration lost?

**Reply by AlexxBlack** [reply]
We have no KMS at all. It's the same server and the same Minio instance. And we don't know how it was working earlier :-)
Maybe You can tell us how to disable encryption in code?
We've found a line:
`ServerSideEncryption: "AES256"`
in the server/utils/s3.js line 110
But commenting of this line trail us to authorization fault without any error message.

**Reply by gempain** [reply]
Happened to me today as I was starting a fresh install. I fixed it by configuring KMS for minion as specified in [their docs in section **2.2 Specify a master key**](https://docs.min.io/docs/minio-vault-legacy.html):

```dotenv
# openssl rand -hex 64
MINIO_KMS_MASTER_KEY=xxxx
```
